lighthouse-core/lib/tracehouse/trace-processor.js

```diff
@@ -524,6 +524,28 @@ class TraceProcessor {
       firstMeaningfulPaint = lastCandidate;
     }
 
+    // LCP comes from the latest `largestContentfulPaint::Candidate`, but it can be invalidated
+    // by a `largestContentfulPaint::Invalidate` event. In the case that the last candidate is
+    // invalidated, the value will be undefined.
+    let largestContentfulPaint;
+    let invalidLcp = false;
+    // Iterate the events backwards.
+    for (let i = frameEvents.length - 1; i >= 0; i--) {
+      const e = frameEvents[i];
+      // If the event's timestamp is before the navigation start, stop.
+      if (e.ts <= navigationStart.ts) break;
+      // If the first lcp event is 'Invalidate', there is inconclusive data to determine LCP.
```

Comment:
```suggestion
      // If the last lcp event in the trace is 'Invalidate', there is inconclusive data to determine LCP.
```
or something like that. Definitely confusing even though traversing backwards (and "last" is used on L544)